service/messageHandler.go

```diff
@@ -0,0 +1,106 @@
+package service
+
+import (
+	"encoding/json"
+	"regexp"
+	"time"
+
+	"github.com/Financial-Times/kafka-client-go/kafka"
+	"github.com/satori/go.uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+const messageTimestampDateFormat = "2006-01-02T15:04:05.000Z"
+
+var predicates = map[string]string{
+	"http://www.ft.com/ontology/classification/isClassifiedBy":"isClassifiedBy",
+	"http://www.ft.com/ontology/annotation/hasAuthor":"hasAuthor",
+	"http://www.ft.com/ontology/annotation/hasContributor":"hasContributor",
+	"http://www.ft.com/ontology/annotation/about":"about",
+	"http://www.ft.com/ontology/annotation/hasDisplayTag":"hasDisplayTag",
+	"http://www.ft.com/ontology/annotation/mentions":"mentions",
+}
+
+type AnnotationMapperService struct {
+	whitelist       *regexp.Regexp
+	messageProducer kafka.Producer
+}
+
+func NewAnnotationMapperService(whitelist *regexp.Regexp, messageProducer kafka.Producer) *AnnotationMapperService {
+	return &AnnotationMapperService{whitelist, messageProducer}
+}
+
+func (mapper *AnnotationMapperService) HandleMessage(msg kafka.FTMessage) error {
+	tid := msg.Headers["X-Request-Id"]
+	requestLog := log.WithField("transaction_id", tid)
+	if mapper.whitelist == nil {
+		requestLog.Error("Skipping this message because the whitelist is invalid.")
```

Comment:
Should the app fail or be unhealthy if the whitelist is invalid? If we do neither, we'll never process any messages, but the app will appear healthy

Reply:
Having said that, we would surely notice that the app wasn't processing any messages in pre-prod / lower envs, so it's probably fine like this.